### PR TITLE
Feature/casting

### PR DESCRIPTION
**Pull request by tvanlaerhoven**
Add chromecast & airplay support

### CODE REVIEW

File: example/ios/ReactNativeTHEOplayer/AppDelegate.m [hunk 1]
```diff
@@ -30,6 +32,13 @@ - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(
   rootViewController.view = rootView;
   self.window.rootViewController = rootViewController;
   [self.window makeKeyAndVisible];
+  
+  NSString *receiverAppID = @"CC1AD845"; // default THEOplayer v3 receiver
```
Comment (Danesz) [inline example/ios/ReactNativeTHEOplayer/AppDelegate.m:36]: should not we parse the `receiverAppID` from the React PlayerConfiguration ?

File: example/src/components/videoplayer/VideoPlayer.tsx [hunk 2]
```diff
@@ -190,6 +200,43 @@ export class VideoPlayer extends PureComponent<VideoPlayerProps, VideoPlayerStat
     console.log(TAG, 'onAdEvent', type, ad);
   };
 
+  private onCastStateChangeEvent = (data: ChromecastChangeEvent | AirplayStateChangeEvent) => {
+    const stateEvent = data;
+    let message = undefined;
+    const castTarget = data.type === 'chromecaststatechange' ? 'chromecast' : 'airplay';
```
Comment (Danesz) [inline example/src/components/videoplayer/VideoPlayer.tsx:206]: nit-pick, can't we check if "data is ChromecastChangeEvent" ?

**Review by Danesz — APPROVED** [review]

File: example/src/components/videoplayer/VideoPlayer.tsx (hunk 2 above)
Comment (tvanlaerhoven) [reply example/src/components/videoplayer/VideoPlayer.tsx:206]: Unfortunately not, both `ChromecastChangeEvent` and `AirplayStateChangeEvent` are interfaces.

File: example/ios/ReactNativeTHEOplayer/AppDelegate.m (hunk 1 above)
Comment (wvanhaevre) [reply example/ios/ReactNativeTHEOplayer/AppDelegate.m:39]: This is an initial default receiver we configure early as suggested bij [(react-native-google-cast)](https://react-native-google-cast.github.io/docs/getting-started/setup). We noticed that when not configured early (e.g. till we bridged the appId), the react-native-google-cast button does not show.
When we receive the bridged playerConfig, we indeed parse the correct receiverAppId and reconfigure the setup
(see [parsing](https://github.com/THEOplayer/react-native-theoplayer/blob/feature/casting/ios/THEOplayerRCTView.swift#:~:text=let%20castReceiverApplicationId%20%3D%20chromecastConfig%5B%22appID%22%5D%20as%3F%20String%20%7B ) end [re-config](https://github.com/THEOplayer/react-native-theoplayer/blob/feature/casting/ios/THEOplayerRCTView.swift#:~:text=let%20options%20%3D%20GCKCastOptions(discoveryCriteria%3A%20GCKDiscoveryCriteria(applicationID%3A%20castReceiverApplicationId)))

**Pull request merged**